rpc/jsonrpc/jsonrpc_test.go

```diff
@@ -265,95 +250,91 @@ func testWithWSClient(ctx context.Context, t *testing.T, cl *client.WSClient) {
 
 //-------------
 
-func TestServersAndClientsBasic(t *testing.T) {
-	// TODO: reenable the leak detector once the test fixture is
-	// managed in the context of this test.
-	//
-	// t.Cleanup(leaktest.Check(t))
-
-	bctx, bcancel := context.WithCancel(context.Background())
-	defer bcancel()
-
-	serverAddrs := [...]string{tcpAddr, unixAddr}
-	for _, addr := range serverAddrs {
-		t.Run(addr, func(t *testing.T) {
-			ctx, cancel := context.WithCancel(bctx)
-			defer cancel()
-
-			logger := log.NewNopLogger()
-
-			cl2, err := client.New(addr)
-			require.NoError(t, err)
-			fmt.Printf("=== testing server on %s using JSONRPC client", addr)
-			testWithHTTPClient(ctx, t, cl2)
-
-			cl3, err := client.NewWS(addr, websocketEndpoint)
-			require.NoError(t, err)
-			cl3.Logger = logger
-			err = cl3.Start(ctx)
-			require.NoError(t, err)
-			fmt.Printf("=== testing server on %s using WS client", addr)
-			testWithWSClient(ctx, t, cl3)
-			cancel()
-		})
-	}
-}
-
-func TestWSNewWSRPCFunc(t *testing.T) {
-	t.Cleanup(leaktest.Check(t))
-
+func TestRPC(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
+	logger := log.NewNopLogger()
 
-	cl, err := client.NewWS(tcpAddr, websocketEndpoint)
-	require.NoError(t, err)
-	cl.Logger = log.NewNopLogger()
-	err = cl.Start(ctx)
-	require.NoError(t, err)
-	t.Cleanup(func() {
-		if err := cl.Stop(); err != nil {
-			t.Error(err)
+	t.Cleanup(leaktest.Check(t))
+	require.NoError(t, setup(ctx, t, logger))
```

Comment:
This changes the setup logger to be the nop logger versus an actual logger. Is that preferred for some reason?

Reply:
I've been slowly moving more tests to use the noop logger (the logging isn't useful during the test debugging cycle, etc, and in practice most tests run with the noop logger because we don't use verbose mode much.)

Having said that, this test has been using the noop logger for a while, the *test* fixture hasn't (because it's started in init rather than in the test, so this is just moving the fixture into the test, and means that it'll get cleaned up ever.)

Reply:
Right, this changes the fixture to use the nop logger, is that preferred?

Reply:
I think so